### ISSUE TITLE
Add support for `grid-template` and `grid` shorthands to `named-grid-areas-alignment`

### ISSUE THREAD

**Issue by danny-huang-openfind** [stylelint-stylistic/stylelint-stylistic#45]
### What is the problem you're trying to solve?

I'm trying to use named-grid-areas-alignment in our repository, but I noticed that when I apply grid-template, this rule no longer takes effect. As a result, the alignment becomes quite messy, especially when using functions like minmax or any function with brackets.

### What solution would you like to see?

I would like the `grid-template` to align as follows:

```css
grid-template:
  "a   a              a     a" auto
  "bb  bb             bb    bb" minmax(0, 1fr)
  "ccc ccc            ccc   ccc" fit-content(0)
/  1fr minmax(0, 1fr) 100px 100px;
```

...or anything that looks visually appealing to most of us. I don't have a specific expectation for it.

**Comment by MorevM** [comment]
Good day @danny-huang-openfind 

Thank you for the issue, I have plans to extend the rule, but it's hard to say when I'll have time to do it.
I personally prefer not to use the shorthand property (I find it harder to perceive than separate ones), so it's irrelevant personally for me, but I can understand those who want to.

Anyway, for now we can discuss what it should look like at least.
Regarding your proposed alignment - it will definitely not be so (the '/' position will conflict with the indentation rule).

I would suggest aligning cell names (if any) and column names (if any), and row sizes (it'll be almost automatic).
The column values, I think, would be pretty hard to align (given the slash position), and the result could be hard to read, so I'd just leave them as is (?), especially since it can also be written on a single line (as far as I can see, this is a popular option).

So, in my opinion, formatted version of your code should look like:

```css
grid-template:
  'a   a   a   a'   auto
  'bb  bb  bb  bb'  minmax(0, 1fr)
  'ccc ccc ccc ccc' fit-content(0)
  / 1fr minmax(0, 1fr) 100px 100px;
```

I'd be happy if you can provide the most contrived and complex (but valid) example property that has all the available options.
Your current example isn't bad, could you use custom row and column names and figure out what that might look like? 
As I said, I have no practical experience with the combined property, nor have a valid example at hand where I could apply this. :)

**Comment by firefoxic** [comment]
> So, in my opinion, formatted version of your code should look like:
> 
> ```css
> grid-template:
>   'a   a   a   a'   auto
>   'bb  bb  bb  bb'  minmax(0, 1fr)
>   'ccc ccc ccc ccc' fit-content(0)
>   / 1fr minmax(0, 1fr) 100px 100px;
> ```

@MorevM  I'm not sure, but it seems like it would be difficult to align parts of the `grid-template-rows` value (the ones after the closing quotes) like in your example. Because it would most likely conflict with the rule that regulates the number of spaces between values. I think that if you want to align these values in a column, you just need to enable the `alignQuotes` option, then the `grid-template-rows` values will be automatically aligned:

```css
grid-template:
  'a   a   a   a  ' auto
  'bb  bb  bb  bb ' minmax(0, 1fr)
  'ccc ccc ccc ccc' fit-content(0)
  / 1fr minmax(0, 1fr) 100px 100px;
```

By the way, I think it's worth providing the option to not move `/` to the next line:

```css
grid-template:
  'a   a   a   a  ' auto
  'bb  bb  bb  bb ' minmax(0, 1fr)
  'ccc ccc ccc ccc' fit-content(0) / 1fr minmax(0, 1fr) 100px 100px;
```

I'm increasingly finding this variant more readable.

---

And lastly — I think it's worth considering the shorthand `grid` at once 😉

**Comment by MorevM** [comment]
> most likely conflict with the rule that regulates the number of spaces between values

Is there such rule? I remember that we [discussed](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/17#issuecomment-2016724375) something like this, but there is no realization yet.

Anyway, the example above looks weird without such alignment. 
`alignQuotes` solves the problem, but firstly, not everyone likes it and secondly, what if there are row names presented?

Something like:

```css
grid-template:
  [header-left] "head head" 30px [header-right]
  [main-left]   "nav  main" 1fr  [main-right]
  [footer-left] "nav  foot" 30px [footer-right]
  / 120px 1fr;
```

Even more complicated (it's not necessary for all tracks to have a name - it's still valid):

```css
grid-template:
  [header-left] "head head" minmax(30px, 1fr) [header-right]
                "nav  main" 1fr               [main-right]
  [footer]      "nav  foot" 30px
  / 120px 1fr;
```

(thats why I dont use shorthand :)) )

Area names can't be aligned without aligning the rest, so I think there should be an exception here.
Also, we'll have a naming issue - it's not only for `named grid areas` in the cases above..

> By the way, I think it's worth providing the option to not move `/` to the next line:

I don't plan to work with it - I'll leave it as written :)

**Comment by firefoxic** [comment]
> I remember that we [discussed](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/17#issuecomment-2016724375) something like this, but there is no realization yet.

Yes, I meant the rule that has not yet been implemented.
